### PR TITLE
CS-1244-punish-collision-fail

### PR DESCRIPTION
**Pull request by CalebSerafin**
## What type of PR is this.
* [x] Bug Fix
* Change
* [x] Enhancement

### What have you changed and why?
Information:
The locality of `_isCollision` was fixed.
Formatting of the hint message has been cleaned up.
fn_oceanGulag.sqf pulls people out of their vehicle for punishment.
Most code runs in an unscheduled environment.

Logging info has been translated from ProgramerSpeak to English:
```log
[Antistasi] | INFO | fn_punishment_FF.sqf | ADMIN, Local Host | FrostsBite [01234567890123456] damaged Petros [AI], Avoided-time: 60, Avoided-offence: 0.4
[Antistasi] | INFO | fn_punishment.sqf | WARNING | FrostsBite [01234567890123456] damaged Petros [AI], Total-time: 60 (incl. +60), Offence+Overhead: 0.4 [0.28+0.12] (incl. +0.4)
[Antistasi] | INFO | fn_punishment_release.sqf | FORGIVE | Player: FrostsBite [01234567890123456]
[Antistasi] | INFO | fn_punishment_release.sqf | RELEASE | Player: FrostsBite [01234567890123456]
```

Special AI are now reported as Victims.
Collateral is now reported in the same line as guilty/warning/exemption:
```log
[Antistasi] | INFO | fn_punishment_FF.sqf | COLLISION | FrostsBite [01234567890123456]'s C_Offroad_01_F damaged Petros [AI]
```

### Please specify which Issue this PR Resolves.
closes [#1244](https://github.com/official-antistasi-community/A3-Antistasi/issues/1244)

### Please verify the following and ensure all checks are completed.

* [x] Have you loaded the mission in LAN host?
* [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
* [ ] No
* Yes

### How can the changes be tested?
Steps: 
1. Shoot someone => warning.
2. Shoot someonex3 => punished.
3. Release explosive/grenade in 75m Petros => grenade notification+warning.
4. Release explosive/grenade in 75m Petrosx3 => punished.
5. Shoot someone from the back of offroad => warning.
6. Shoot someone from the back of offroadx3 => pulled out of vic+punished.
7. Shoot someone from crew-served gun => warning.
8. Shoot someone from crew-served gunx3 => pulled out of vic+punished.
9. Slowly hurt someone from collision =>  collision notification+warning.
10. Kill someone from collision =>  collision notification.
11. Hurt someone from collisionx3 =>  pulled out of vic+punished.

### CODE REVIEW

**Comment by HakonRydland** [comment]
from testing today, ff was always considered from a vehicle. and punishment is never performed anymore.

**Comment by CalebSerafin** [comment]
Bug: collision is grabbing the playable unit as a vehicle.

**Comment by CalebSerafin** [comment]
> Bug: collision is grabbing the playable unit as a vehicle.

Nope, must be another issue during testing. Maybe lag.

**Comment by CalebSerafin** [comment]
Problem discovered: Source usually contains the unit itself even outside of collisions. The original issue was that locality of _isCollision was incorrect. Further testing is not needed.

**Comment by Bob-Murphy** [comment]
@HakonRydland please re-review

**Comment by HakonRydland** [comment]
still waiting for the discuses changes

File: A3-Antistasi/functions/Punishment/fn_punishment_oceanGulag.sqf [hunk 1]
```diff
@@ -69,6 +68,8 @@ switch (toLower _operation) do {
 		_punishment_platform setPos [_pos2D #0, _pos2D #1, -0.25];
 
 		if (isPlayer _detainee) then {
+			// TODO Pull people from vehicles
+			if !(isNull objectParent _detainee) then { moveOut _detainee };
```
[the remark this answers is not in the document]
Comment (HakonRydland) [reply A3-Antistasi/functions/Punishment/fn_punishment_oceanGulag.sqf:72]: if vehicle is stationary it also places the unit back in the vehicle (same pos not in a seat)

**Pull request merged**